contracts/TwabRewards.sol

```diff
@@ -0,0 +1,451 @@
+        return _getPromotion(_promotionId);
+    }
+
+    /// @inheritdoc ITwabRewards
+    function getRemainingRewards(uint256 _promotionId) external view override returns (uint256) {
+        return _getRemainingRewards(_promotionId);
+    }
+
+    /// @inheritdoc ITwabRewards
+    function getCurrentEpoch(uint256 _promotionId) external view override returns (Epoch memory) {
+        return _getCurrentEpoch(_promotionId);
+    }
+
+    /// @inheritdoc ITwabRewards
+    function getEpoch(uint256 _epochId, uint256 _promotionId)
+        external
+        view
+        override
+        returns (Epoch memory)
+    {
+        Promotion memory _promotion = _getPromotion(_promotionId);
+
+        return _getEpoch(_epochId, _promotion);
+    }
+
+    /// @inheritdoc ITwabRewards
+    function getRewardAmount(
+        address _user,
+        uint256 _promotionId,
+        uint256 _epochId
+    ) external view override returns (uint256) {
+        return _calculateRewardAmount(_user, _promotionId, _epochId);
+    }
+
+    /// @inheritdoc ITwabRewards
+    function claimRewards(
+        address _user,
+        uint256 _promotionId,
+        uint256 _epochId
+    ) external override returns (uint256) {
+        require(_epochId < _getCurrentEpoch(_promotionId).id, "TwabRewards/epoch-not-over");
+        require(
+            !_isClaimedEpoch(_user, _promotionId, _epochId),
+            "TwabRewards/rewards-already-claimed"
+        );
+
+        uint256 _rewardAmount = _calculateRewardAmount(_user, _promotionId, _epochId);
+
+        IERC20 _token = IERC20(_getPromotion(_promotionId).token);
+        _token.safeTransferFrom(address(this), _user, _rewardAmount);
+
+        _setClaimedEpoch(_claimedEpochs[_promotionId][_user], _epochId, true);
+
+        emit RewardsClaimed(_token, _rewardAmount);
+
+        return _rewardAmount;
+    }
+
+    /* ============ Internal Functions ============ */
+
+    /**
+    @notice Determine if address passed is actually a ticket.
+    @param _ticket Address to check
+   */
+    function _requireTicket(address _ticket) internal view {
+        require(address(_ticket) != address(0), "TwabRewards/ticket-not-zero-address");
+
+        (bool succeeded, bytes memory data) = address(_ticket).staticcall(
+            abi.encodePacked(ITicket(_ticket).controller.selector)
+        );
+
+        address controllerAddress;
+
+        if (data.length > 0) {
+            controllerAddress = abi.decode(data, (address));
+        }
+
+        require(succeeded && controllerAddress != address(0), "TwabRewards/invalid-ticket");
+    }
+
+    /**
+        @notice Determine if the number of epochs passed exceeds the maximum number of epochs.
+        @param _numberOfEpochs Number of epochs to check
+    */
+    function _requireEpochLimit(uint256 _numberOfEpochs) internal pure {
+        require(_numberOfEpochs < 256, "TwabRewards/exceeds-256-epochs-limit");
```

Comment:
you could even get more specific here and go:

`_numberOfPicks <= type(uint8).max`

Reply:
Done: https://github.com/pooltogether/v4-periphery/pull/8/commits/a84185a0f653742081a84bbdaf95843b2dedd370